sql/_13_issues/_24_1h/cases/cbrd_25195.sql

```diff
@@ -0,0 +1,319 @@
+-- This test case verifies the following issue: CBRD-25195.
+-- A core dump occurs when multiple tables and serial functions are used simultaneously in one query.
+-- Ensure the core is no longer dumped and the expected query result is returned
+
+-- Create 20 tables
+drop table if exists t1;
+drop table if exists t2;
+drop table if exists t3;
+drop table if exists t4;
+drop table if exists t5;
+drop table if exists t6;
+drop table if exists t7;
+drop table if exists t8;
+drop table if exists t9;
+drop table if exists t10;
+drop table if exists t11;
+drop table if exists t12;
+drop table if exists t13;
+drop table if exists t14;
+drop table if exists t15;
+drop table if exists t16;
+drop table if exists t17;
+drop table if exists t18;
+drop table if exists t19;
+drop table if exists t20;
+create table t1 (a int);
+create table t2 (a int);
+create table t3 (a int);
+create table t4 (a int);
+create table t5 (a int);
+create table t6 (a int);
+create table t7 (a int);
+create table t8 (a int);
+create table t9 (a int);
+create table t10 (a int);
+create table t11 (a int);
+create table t12 (a int);
+create table t13 (a int);
+create table t14 (a int);
+create table t15 (a int);
+create table t16 (a int);
+create table t17 (a int);
+create table t18 (a int);
+create table t19 (a int);
+create table t20 (a int);
+
+--/* ----------------------------------------
+--*
+--* Test case 1 - min_tables_single_cast
+--* 
+--* Ensure core dump no longer occurs when using 10 tables and 1 serial function.
+--*     => Selects data from 9 tables -> select 1 serial function -> select 10th table;
+--* Expected Result: The query should return 1
+--* ---------------------------------------- */
+create serial sr1;
+
+select a from t1
+union all
+select a from t2
+union all
+select  a from t3
+union all
+select  a from t4
+union all
+select  a from t5
+union all
+select  a from t6
+union all
+select a from t7
+union all
+select a from t8
+union all
+select a from t9
+union all
+select cast(sr1.next_value as int)
+union all
+select a from t10;
+
+drop serial sr1;
+
+
+--/* ----------------------------------------
+--*
+--* Test case 2 - min_tables_many_casts
+--* 
+--* Ensure core dump no longer occurs when using 10 tables and 10 serial functions sequentially.
+--*     => Selects data from 10 tables and 10 serial functions sequentially and returns the number of rows in the output 
+--* Expected Result: The query should return 10 results with values 1 
+--* ---------------------------------------- */
```

Comment:
select 'Test case 2 - min_tables_many_casts';

Reply:
I only checked it in Test case 1.
1) it has nothing to do with next_values and current_values
- union all
- select cast(sr1.CURRENT_VALUE as int) 
2) It's not related to setting AUTO_INCREMENT
- create table t1 (a int AUTO_INCREMENT, id varchar);
3) It only happens in the 10th.
4) Also occurs if UNION

Please consider adding a case 3, 4

Reply:
The suggested changes/additions have been reflected in the latest commit:
1. it has nothing to do with next_values and current_values ->  'Test case 6 - current_value_cast'
2. It's not related to setting AUTO_INCREMENT -> 'Test case 7 - auto_increment_table_cast'
3. It only happens in the 10th -> Test case 2,3,4
4. Also occurs if UNION -> 'Test case 5 - distinct_union_cast'